main.js

```diff
@@ -72,25 +78,27 @@ function savePalette() {
 function displaySavedPalettes() {
 	miniPaletteContainer.innerHTML = '';
 	for (var i = 0; i < savedPalettes.length; i++) {
-			var miniPalette = document.createElement('section');
-			miniPalette.className = 'mini-palette';
-			for (var j = 0; j < savedPalettes[i].length; j++) {
-					var miniBox = document.createElement('div');
-					miniBox.className = 'mini-box';
-					miniBox.style.backgroundColor = savedPalettes[i][j];
-					miniPalette.appendChild(miniBox);
-			}
-			var deleteButton = document.createElement('button');
-			deleteButton.className = 'delete-button';
-			deleteButton.dataset.index = i;
-			deleteButton.innerHTML = `<img src="assets/delete.png" alt="Delete Button">`;
-			miniPalette.appendChild(deleteButton);
-			miniPaletteContainer.appendChild(miniPalette);
+		var miniPalette = document.createElement('section');
+		miniPalette.className = 'mini-palette';
+		miniPalette.dataset.index = i;
+		for (var j = 0; j < savedPalettes[i].length; j++) {
+			var miniBox = document.createElement('div');
+			miniBox.className = 'mini-box';
+			miniBox.dataset.index = i;
+			miniBox.style.backgroundColor = savedPalettes[i][j];
+			miniPalette.appendChild(miniBox);
+		}
+		var deleteButton = document.createElement('button');
+		deleteButton.className = 'delete-button';
+		deleteButton.dataset.index = i;
```

Comment:
In this block of code (81-96) we utilize dataset.index again in order to assign index positions to the miniBoxes, miniPalettes, and delete button. Makes the code dynamic!